### ISSUE TITLE
add_MARBL: introduce _USE_MARBL_TRACERS CPP macro

### ISSUE THREAD

**Issue by mnlevy1981** [mnlevy1981/MOM6#21]
#22 did the following:

1. Remove MARBL from git submodules
2. Introduce MARBL into Externals_MOM.cfg

This issue will continue the work:

3. Wrap the bulk of `MARBL_tracers.F90` and all calls into the module from `MOM_tracer_control_flow.F90` in `#ifdef USE_MARBL_TRACERS`
4. Make sure MOM builds without `_USE_MARBL_TRACERS`
5. Update CESM build to include always include CPP

**Comment by mnlevy1981** [comment]
An interesting thought from Matt L: if the CPP is available, we should use it in the CESM build. Meaning compsets that are not set up to run with MARBL should not build MARBL.

1. This effectively makes MARBL a build-time specification despite tracer count being run-time (but how often do CESM users decide to turn on a tracer package by hand rather than via compset?)
2. What will be the mechanism for turning MARBL on by default via compset? It seems like `MOM%MARBL` in the compset long name should set an `env_build.xml` variable - does MOM still use `OCN_TRACER_MODULES` or do we need something new like `MOM_ENABLE_MARBL`?

**Comment by mnlevy1981** [comment]
Given how this work is breaking down, I think https://github.com/mnlevy1981/MOM6/issues/21#issuecomment-621427372 should be addressed in a separate issue once this work is complete

**Comment by mnlevy1981** [comment]
Having trouble triggering the abort when running with `USE_MARBL_TRACERS = True` and not building with `-D_USE_MARBL_TRACERS`, need to track call to `call_tracer_flux_init()` [that's where the generic tracer abort is found]

**Comment by mnlevy1981** [comment]
It turns out that `call_tracer_register()` is a better place for the error check (may eventually end up in both routines?). With some CIME mods to add the `-D_USE_MARBL_TRACERS` cpp, this issue is resolved
